extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -478,4 +524,196 @@ void registerServiceBinding(Capabilities capabilities,
                             KafkaBindingConverter.class.getName()));
         }
     }
+
+    // Kafka UI related stuff
+
+    @BuildStep
+    public AdditionalBeanBuildItem kafkaAdminClient() {
+        return AdditionalBeanBuildItem.builder()
+                .addBeanClass(KafkaAdminClient.class)
+                .setUnremovable()
+                .build();
+    }
+
+    @BuildStep
+    public AdditionalBeanBuildItem kafkaTopicClient() {
+        return AdditionalBeanBuildItem.builder()
+                .addBeanClass(KafkaTopicClient.class)
+                .setUnremovable()
+                .build();
+    }
+
+    @BuildStep
+    public AdditionalBeanBuildItem kafkaWebUiUtils() {
+        return AdditionalBeanBuildItem.builder()
+                .addBeanClass(KafkaUiUtils.class)
+                .setUnremovable()
+                .build();
+    }
+
+    @Record(ExecutionTime.RUNTIME_INIT)
+    @BuildStep
+    @Consume(BeanContainerBuildItem.class)
+    void buildExecutionEndpoint(
+            BuildProducer<RouteBuildItem> routeProducer,
+            HttpRootPathBuildItem httpRootPathBuildItem,
+            KafkaUiRecorder recorder,
+            ShutdownContextBuildItem shutdownContext,
+            LaunchModeBuildItem launchMode,
+            BodyHandlerBuildItem bodyHandlerBuildItem,
+            KafkaBuildTimeConfig buildConfig,
+            BeanContainerBuildItem beanContainer,
+            BuildProducer<WebsocketSubProtocolsBuildItem> webSocketSubProtocols) {
+
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public void registerKafkaUiExecHandler(
+            BuildProducer<RouteBuildItem> routeProducer,
```

Comment:
No because this is not really Dev UI (It's a UI that is linked from Dev UI, like Swagger UI) You can not run a DevConsoleRoute outside of dev mode, where with this UI, you can enable that with a config (again like Swagger UI)